middleware/index.ts

```diff
@@ -188,21 +223,11 @@ export default async function systemFn(message: any): Promise<any> {
             throw new MiddlewareError(parseErr, 400);
         }
 
-        let event: InvocationEvent;
-        let context: Context;
-        let logger: Logger;
-        try {
-            // Create function param objects from request
-            [event, context, logger] = applySfFnMiddleware(cloudEvent, headers, requestLogger);
-        } catch (apiSetupError) {
-            throw new MiddlewareError(apiSetupError);
-        }
-
         // Invoke requested function
         let result: any;
         const startExecTimeMs = new Date().getTime();
         try {
-            result = await userFn(...[event, context, logger]);
+            result = await userFn(cloudEvent.toJSON(), headers);
```

Comment:
I'm a little worried that this will create a *mixed* cloudEvent --

**Concern**: The toJSON method will return a *structured* cloudevent object (https://github.com/cloudevents/sdk-javascript/blob/main/src/event/cloudevent.ts#L142), yet the `headers` will (I think?) still contain the HTTP Binary encoding `ce-id`, `ce-type`, ... headers.  Not sure how to resolve yet, but I think it would involve stripping out any `ce-*` headers and setting `Content-Type` header from `cloudEvent.datacontenttype` if available.

Reply:
Yes, the `headers` will remain untouched and still have `ce-*` values. We probably should update this code a bit to be more clear. We aren't concerned with that duplication since the consumer is the sdk. Is there a reason you think we should be more concerned? The SDK currently doesn't do anything with headers in relation to the cloudevent values.

While we are using the `cloudEvent`, we made an explicit interface between the middleware and the sdk a subset of that. We may be providing additional keys, but they aren't part of the consumed contract.

I might take a stab at explicitly turning a cloudevent into the this smaller interface instead of running with the assumption that `toJSON()` will always produce the minimum matching contract.

This is in the interface we are expecting the middleware to produce.
```
interface SdkCloudEvent {
    id:               string,
    type:             string,
    data:             JSON,
    source:           string,
    specversion:      string,
    datacontenttype?: string,
    sfcontext?:       any,
    sffncontext?:     any,
    schemaurl?:       string,
    time?:            string,
}
```